src/run_task.py

```diff
@@ -125,58 +145,104 @@ def main(
+            )
+
+            itempath = S3ItemPath(
+                bucket=output_bucket,
+                sensor="s2",
+                dataset_id="vegheight",
+                version=version,
+                time=datetime,
+            )
+
+            searcher = PystacSearcher(
+                catalog=catalog,
+                collections=[collection],
+                datetime=datetime,
+            )
+
+            loader = OdcLoader(
+                bands=["B04", "B03", "B02", "observations"],  # , "B08"],
+                chunks={"x": 9600, "y": 9600},
+            )
+            items = searcher.search(geobox)
+            if len(items) == 0:
+                log.info(f"Image not found for this date {datetime}")
+                continue
+
+            # tile_id is a string like "45,55"
+            stac_document = itempath.stac_path(tile_id)
+            log.info(
+                f"result will be saved to https://{output_bucket}.s3.us-west-2.amazonaws.com/{stac_document}"
+            )
+
+            # If we don't want to overwrite, and the destination file already exists, skip it
+            if not overwrite and object_exists(
+                output_bucket, stac_document, client=client
+            ):
+                log.info(f"Item already exists at {stac_document}")
+                # This is an exit with success
+                raise typer.Exit()
+
+            # Make sure we pass original client down to s3 writer and stac writer
+            dep_client_to_s3 = partial(write_to_s3, client=client)
+            # Custom writer so we write multithreaded
+            writer = AwsDsCogWriter(
+                itempath, write_multithreaded=True, write_function=dep_client_to_s3
+            )
+            # STAC making thing
+            stac_creator = StacCreator(
+                itempath=itempath,
+                collection_url_root=collection_url_root,
+                remote=True,
+                make_hrefs_https=True,
+                with_raster=True,
+            )
+            stac_writer = CustomAwsStacWriter(
+                itempath=itempath, write_stac_function=dep_client_to_s3
+            )
+
+            try:
+                paths = Task(
+                    itempath=itempath,
+                    id=tile_index,
+                    area=geobox,
+                    searcher=searcher,
+                    loader=loader,
+                    processor=processor,
+                    writer=writer,
+                    logger=log,
+                    stac_creator=stac_creator,
+                    stac_writer=stac_writer,
+                ).run()
+            except EmptyCollectionError:
+                log.info("No items found for this tile")
+                raise typer.Exit()  # Exit with success
+            except Exception as e:
+                log.exception(f"Failed to process with error: {e}")
+                raise typer.Exit(code=1)

```

Comment:
These exceptions will exit the loop of dates, so all future dates will fail.

We're better off handling the multiple dates in the "print tasks" stage of this work.